# NET PROMOTING SCORE (NPS):
It's basically the '%' of promoter which are satisfied by service/product out of total

In [1]:
import numpy as np

In [2]:
score = np.loadtxt("survey.txt",dtype='int')
score

array([ 7, 10,  5, ...,  5,  9, 10])

## MAX SCORE

In [3]:
np.max(score)

10

# MIN SCORE

In [4]:
np.min(score)

1

# RANGE OF SCORE

**THE RANGE LISE BETWEEN 1 & 10**

## TOTAL NO OF FEEDBACKS

In [5]:
#total no of feedbacks
total_people = score.shape[0]
total_people

1167

## PROMOTERS
**PROMOTERS: The no of people who are satisfied with service and willing to promote(score >= 8)**

In [6]:
promoters = score[score > 8]
promoters_count = promoters.shape[0]
promoters_count

609

## DETRACTORS
**DETRACTORS: The no of people who are not satisfied with service or having a bad 
expeience and not willing to promote(score < 6)**

In [7]:
detractors = score[score < 6]
detractors_count = detractors.shape[0]
detractors_count

332

## NEUTRALS
**NEUTRALS: The no of people who are not interseted in feedback(score >= 6 and score < 8)**

In [8]:
neutrals = score[(score >= 6) & (score <= 8)]
neutrals_count = neutrals.shape[0]
neutrals_count

226

# NPS FORMULA
**nps = (%promoters) - (% detractors)**

In [9]:
# percentage of promoters
percentage_promoters = ((promoters_count * 100) / total_people)
print('percentage of promoters :',percentage_promoters,'%',sep=' ')

# percentage of detractors
percentage_detractors = ((detractors_count * 100) / total_people)
print('percentage of detractors :',percentage_detractors,'%',sep=' ')

#calculations of NPS
NPS = percentage_promoters - percentage_detractors
print("NPS of AIRBNB's is : " ,NPS,'%',sep=' ')

percentage of promoters : 52.18508997429306 %
percentage of detractors : 28.449014567266495 %
NPS of AIRBNB's is :  23.73607540702656 %


In [10]:
np.round(NPS,2)

23.74

# CONCLUSION

> THE NPS IS 23.74 % WHICH MEAN ITS A POSITIVE FEEDBACK & LOTS OF CUSTOMER ARE SATISFIED WITH THEIR SERVICE FOR THIS DATA